config/permissions.cfg

```diff
@@ -11,12 +12,26 @@
 # Parameters:
 # -----------
 #   service:              service name to receive the permission (directly on it if no 'resource' mentioned, must exist)
-#   resource (optional):  tree path of the service's resource (ex: /res1/sub-res2/sub-sub-res3)
-#   type (optional):      resource type employed in case of ambiguity between multiple choices (depends on service type)
-#   user and/or group:    user/group for which to apply the permission (create if missing, see below)
+#   resource:  (optional) tree path of the service's resource (ex: /res1/sub-res2/sub-sub-res3)
+#   type:      (optional) resource type employed in case of ambiguity between multiple choices (depends on service type)
+#   user:                 user for which to apply the modification of permission (skip if omitted or None)
+#   group:                group for which to apply the modification of permission (skip if omitted or None)
 #   permission:           name or object of the permission to be applied (see 'magpie.permissions' for supported values)
 #   action:               one of [create, remove] (default: create)
 #
+# User/Group:
+# -----------
+#   At least one of [user, group] must be provided for every permission entry specified to create/remove.
+#   Both can also be specified simultaneously to create/remove the same permission for both cases.
+#
+#   Whenever a specified user/group doesn't exist in the database for permission creation, it gets created dynamically.
+#   When this dynamic creation must occur, sections 'users' and 'groups' are looked for possible extended definitions,
+#   as presented in 'config.yml'. Otherwise, the corresponding defaults are employed when no match was found.
+#
+#   The 'users' and 'groups' sections must be at the top-level of the configuration (same as 'permissions').
+#   This is to allow reuse of the same named reference multiple times across many permissions operations without
+#   needing to repeat the whole user/group parameters each time.
```

Comment:
And what happens if it's not the case? Simply repeat the user/group parameters each time, or might have undesired behaviour?

Reply:
It merges all `users` and all `groups` in every matched config. 
Then, using those mappings, attempts to resolve corresponding `user` and group` references. 
If matched, it uses the full definition. Otherwise, the plain string for `name` + other fields defaults. 

If this definition clear enough or needs more details? I will add is as is if it helps understand.

Reply:
The explanation makes sense, but maybe just put the "otherwise" section after the phrase that states "must". 
Usually when reading something like "there must be", we seek to find a "and otherwise, this would happen".
If it's only to avoid repeating the user/group params each time, maybe just state that it would be the only consequence.

Reply:
I pushed another update, let me know if it is clearer.

Reply:
Yes, thanks for the update.